### PR TITLE
Refactor oauth to use google via omniauth rather than hand rolled oauth with ynab

### PR DESCRIPTION
**Pull request by grwthomps**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -4,29 +4,30 @@ def new
...
+  def user_data
     {
-      access_token: raw_token_data[:access_token],
-      refresh_token: raw_token_data[:refresh_token],
-      last_login: DateTime.now
+      email: request.env['omniauth.auth']["info"]["email"],
+      first_name: request.env['omniauth.auth']["info"]["first_name"],
+      last_name: request.env['omniauth.auth']["info"]["last_name"],
+      image: request.env['omniauth.auth']["info"]["image"],
+      google_token: request.env['omniauth.auth']["credentials"]["token"]
```
Comment (zacisaacson) [inline app/controllers/users_controller.rb]: Look great!

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -4,29 +4,30 @@ def new
...
-    session[:user_id] = user.id
-    redirect_to signup_profile_path
-  end
-
-  def update
-    current_user.update!(user_params)
-    redirect_to signup_accounts_path
-  end
-
-  def edit
```
Comment (hillstew) [inline app/controllers/users_controller.rb]: We could probably break this out into some helper methods as a refactor

File: app/controllers/ynab_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def create
+    tokens = YnabService.new.fetch_tokens(params['code'])
+    user_data = format_user_data(tokens)
+    current_user.update!(user_data)
+    redirect_to signup_accounts_path
+  end
+
+  private
+
+  def format_user_data(raw_token_data)
```
Comment (hillstew) [inline app/controllers/ynab_controller.rb]:
```suggestion
  def format_user_data(token_data)
```

File: app/controllers/ynab_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {
+      access_token: raw_token_data[:access_token],
```
Comment (hillstew) [inline app/controllers/ynab_controller.rb]:
```suggestion
      access_token: token_data[:access_token],
```

File: app/controllers/ynab_controller.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      refresh_token: raw_token_data[:refresh_token],
```
Comment (hillstew) [inline app/controllers/ynab_controller.rb]:
```suggestion
      refresh_token: token_data[:refresh_token],
```

File: config/routes.rb [hunk 6]
```diff
@@ -1,13 +1,16 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  get '/', to: 'welcome#index'
+  get '/', to: 'welcome#index', as: 'welcome'
   get '/signup', to: 'users#new'
-  get '/auth/ynab/callback', to: 'users#create'
+  get '/auth/ynab/callback', to: 'ynab#create'
```
Comment (lrs8810) [inline config/routes.rb]: We can clean up these routes to be grouped together (i.e. ynab, users, etc.)

**Pull request merged**